app/controllers/concerns/mfa_setup_concern.rb

```diff
@@ -2,6 +2,16 @@ module MfaSetupConcern
   extend ActiveSupport::Concern
 
   def user_next_authentication_setup_path!(final_path = nil)
+    if user_session.dig(:selected_mfa_options).present? &&
+       IdentityConfig.store.select_multiple_mfa_options
+      auth_method_confirmation_url(final_path: final_path)
+    else
+      user_session[:selected_mfa_options] = nil
+      final_path
+    end
+  end
+
+  def confirmation_path(final_path = nil)
```

Comment:
looks like `user_next_authentication_setup_path!` no longer mutates the session but `confirmation_path` does so maybe we move the `!` from one to the other?